ssz/sedes/serializable.py

```diff
@@ -0,0 +1,419 @@
+
+    def __copy__(self):
+        return self.copy()
+
+    def __deepcopy__(self, *args):
+        return self.copy()
+
+    _in_mutable_context = False
+
+
+def make_immutable(value):
+    if isinstance(value, list):
+        return tuple(make_immutable(item) for item in value)
+    else:
+        return value
+
+
+@to_tuple
+def _mk_field_attrs(field_names, extra_namespace):
+    namespace = set(field_names).union(extra_namespace)
+    for field in field_names:
+        while True:
+            field = '_' + field
+            if field not in namespace:
+                namespace.add(field)
+                yield field
+                break
+
+
+def _mk_field_property(field, attr):
+    def field_fn_getter(self):
+        return getattr(self, attr)
+
+    def field_fn_setter(self, value):
+        if not self._in_mutable_context:
+            raise AttributeError("can't set attribute")
+        setattr(self, attr, value)
+
+    return property(field_fn_getter, field_fn_setter)
+
+
+IDENTIFIER_REGEX = re.compile(r"^[^\d\W]\w*\Z", re.UNICODE)
+
+
+def _is_valid_identifier(value):
+    # Source: https://stackoverflow.com/questions/5474008/regular-expression-to-confirm-whether-a-string-is-a-valid-identifier-in-python  # noqa: E501
+    if not isinstance(value, str):
+        return False
+    return bool(IDENTIFIER_REGEX.match(value))
+
+
+@to_set
+def _get_class_namespace(cls):
+    if hasattr(cls, '__dict__'):
+        yield from cls.__dict__.keys()
+    if hasattr(cls, '__slots__'):
+        yield from cls.__slots__
+
+
+class SerializableBase(abc.ABCMeta):
+    def __new__(cls, name, bases, attrs):
+        super_new = super(SerializableBase, cls).__new__
+
+        serializable_bases = tuple(b for b in bases if isinstance(b, SerializableBase))
+        has_multiple_serializable_parents = len(serializable_bases) > 1
+        is_serializable_subclass = any(serializable_bases)
+        declares_fields = 'fields' in attrs
+
+        if not is_serializable_subclass:
+            # If this is the original creation of the `Serializable` class,
+            # just create the class.
+            return super_new(cls, name, bases, attrs)
+        elif not declares_fields:
+            if has_multiple_serializable_parents:
+                raise TypeError(
+                    "Cannot create subclass from multiple parent `Serializable` "
+                    "classes without explicit `fields` declaration."
+                )
+            else:
+                # This is just a vanilla subclass of a `Serializable` parent class.
+                parent_serializable = serializable_bases[0]
+
+                if hasattr(parent_serializable, '_meta'):
+                    fields = parent_serializable._meta.fields
+                else:
+                    # This is a subclass of `Serializable` which has no
+                    # `fields`, likely intended for further subclassing.
+                    return super_new(cls, name, bases, attrs)
```

Comment:
I don't know how to fix it and it's not critical, but I'd like to point out that this line seems to be the root of this issue: https://github.com/ethereum/pyrlp/issues/101

Reply:
Fixed

Reply:
It's working fine after I added support for classes which are declared with `empty fields` (reference https://github.com/ethereum/pyrlp/issues/102). 

Attaching the screenshot for reference.
![screenshot from 2018-12-24 11-34-23](https://user-images.githubusercontent.com/15922456/50391970-17e31380-0770-11e9-9c79-d70994c39179.png)

Reply:
Sorry, just declared the class but didn't instantiate it. Pushed another commit for that.